hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -493,6 +491,15 @@ int MDMParser::waitFinalResp(_CALLBACKPTR cb /* = NULL*/,
                             else if (a == 3) *reg = REG_DENIED;   // 3: registration denied
                             else if (a == 4) *reg = REG_UNKNOWN;  // 4: unknown
                             else if (a == 5) *reg = REG_ROAMING;  // 5: registered, roaming
+                            if (_dev.dev == DEV_SARA_R410 && _attached && !strcmp(s, "CEREG:") && *reg == REG_NONE) {
+                                MDM_PRINTF("Cell reg disconnected\r\n");
+                                _attached_urc = (*reg == REG_NONE) ? 0:1;
+                                if (!_attached_urc) { // Regitration detected as 0 or 2
+                                    _ip = NOIP;
+                                    _attached = false;
+                                    HAL_NET_notify_disconnected();
+                                }
+                            }
```

Comment:
Nit: spelling on "Registration"

Should `&& *reg == REG_NONE` be `&& (*reg != REG_HOME || *reg != REG_ROAMING))` since that's what we define as `REG_OK()`

`_attached_urc = (*reg == REG_NONE) ? 0:1;` would also not need to be tested, just set to 0, and also then no need to test for `if (!_attached_urc)`.

Perhaps also put the `&& !strcmp(s, "CEREG:")` last, as it will chew up more cycles.  

We can also test `&& reg == &_net.eps` instead of strcmp() again?